packages/flutter_tools/lib/src/migrate/migrate_utils.dart

```diff
@@ -0,0 +1,325 @@
+    return true;
+  }
+
+  /// Returns true if the workingDirectory is a git repo.
+  static Future<bool> isGitRepo(String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['rev-parse', '--is-inside-work-tree'];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[-1], commandDescription: 'git ${cmdArgs.join(' ')}');
+    if (result.exitCode == 0) {
+      return true;
+    }
+    return false;
+  }
+
+  /// Returns true if the file at `filePath` is covered by the `.gitignore`
+  static Future<bool> isGitIgnored(String filePath, String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['check-ignore', filePath];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[0, 1, 128], commandDescription: 'git ${cmdArgs.join(' ')}');
+    return result.exitCode == 0;
+  }
+
+  /// Runs `flutter pub upgrate --major-revisions`.
+  static Future<void> flutterPubUpgrade(String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['pub', 'upgrade', '--major-versions'];
+    final ProcessResult result = await Process.run('flutter', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[0], commandDescription: 'flutter ${cmdArgs.join(' ')}');
+  }
+
+  /// Runs `./gradlew tasks` in the android directory of a flutter project.
+  static Future<void> gradlewTasks(String workingDirectory, Logger logger) async {
+    final String baseCommand = Platform.isWindows ? 'gradlew.bat' : './gradlew';
+    final List<String> cmdArgs = <String>['tasks'];
+    final ProcessResult result = await Process.run(baseCommand, cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[0], commandDescription: '$baseCommand ${cmdArgs.join(' ')}');
+  }
+
+  /// Verifies that the ProcessResult does not contain an error.
+  ///
+  /// If an error is detected, the error can be optionally logged or exit the tool.
+  static bool checkForErrors(
+    ProcessResult result,
+    Logger logger, {
+    List<int> allowedExitCodes = const <int>[],
+    String? commandDescription,
+    bool exit = true,
+    bool silent = false
+  }) {
+    // -1 in allowed exit codes means all exit codes are valid.
+    if ((result.exitCode != 0 && !allowedExitCodes.contains(result.exitCode)) && !allowedExitCodes.contains(-1)) {
+      if (!silent) {
+        logger.printError('Command encountered an error with exit code ${result.exitCode}.');
+        if (commandDescription != null) {
+          logger.printError('Command:');
+          logger.printError(commandDescription, indent: 2);
+        }
+        logger.printError('Stdout:');
+        logger.printStatus(result.stdout as String, indent: 2);
```

Comment:
This is still an error diagnostic

```suggestion
        logger.printError(result.stdout as String, indent: 2);
```